src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Triggers/EventHubTriggerInput.cs

```diff
@@ -51,6 +52,45 @@ public EventHubTriggerInput GetSingleEventTriggerInput(int idx)
         public EventData GetSingleEventData()
         {
             return this.Events[this._selector];
-        }        
+        }
+
+        public Dictionary<string, string> GetTriggerDetails(PartitionContext context)
+        {
+            if (IsSingleDispatch)
+            {
+                return new Dictionary<string, string>()
+                {
+                    { "PartionId", context.PartitionId },
+                    { "Offset", string.Join(",", Events[0].SystemProperties?.Offset) },
+                    { "EnqueueTimeUtc", string.Join(",", Events[0].SystemProperties?.EnqueuedTimeUtc.ToString("o")) },
+                    { "PartitionKey", string.Join(",", Events[0].SystemProperties?.PartitionKey) },
+                    { "SequenceNumber", string.Join(",", Events[0].SystemProperties?.SequenceNumber.ToString()) }
+                };
+            }
+            else
+            {
+                string[] offsetArray = new string[Events.Length];
+                string[] enqueueTimeUtcArray = new string[Events.Length];
+                string[] partitionKeyArray = new string[Events.Length];
+                string[] sequenceNumberArray = new string[Events.Length];
+
+                for (int i = 0; i < Events.Length; i++)
+                {
+                    offsetArray[i] = Events[i].SystemProperties?.Offset;
+                    enqueueTimeUtcArray[i] = Events[i].SystemProperties?.EnqueuedTimeUtc.ToString("o");
+                    partitionKeyArray[i] = Events[i].SystemProperties?.PartitionKey;
+                    sequenceNumberArray[i] = Events[i].SystemProperties?.SequenceNumber.ToString();
+                }
+
+                return new Dictionary<string, string>()
+                {
+                    { "PartionId", context.PartitionId },
+                    { "OffsetArray", string.Join(",", offsetArray) },
+                    { "EnqueueTimeUtcArray", string.Join(",", enqueueTimeUtcArray) },
+                    { "PartitionKeyArray", string.Join(",", partitionKeyArray) },
+                    { "SequenceNumberArray", string.Join(",", sequenceNumberArray) }
+                };
+            }
+        }
```

Comment:
This could get pretty large if the batchsize is big, right? Anything to be concerned about with this?